src/compute/aggregate/sum.rs

```diff
@@ -19,23 +19,43 @@ pub trait Sum<T> {
     fn simd_sum(self) -> T;
 }
 
+fn split_by_alignment<U, T>(values: &[T]) -> (&[T], &[T]) {
```

Comment:
I would expect this method to have signature `fn split_by_alignment<U, T>(values: &[T]) -> (&[T], &[T], &[T])` to also take care of the tail. An option here is to use [align_to](https://doc.rust-lang.org/std/primitive.slice.html#method.align_to), which may lead to not needing `_unchecked`

Reply:
I like that one, because then we don't need the extra trait method anymore. As this leaves the middle chunk already of the `Simd` type.